Overwrite and other things added

AlexLi0104 opened the pull request:
@cboettig 

Greetings! I have modified the `download_gpdd` function so that it takes in an overwrite argument,  with default being "Ask". I also added a message showing the downloading process. I modified the Readme and vignettes as well. Please feel free to take a look and let me know what else needs to be changed. 

Thank you!

cboettig commented:
Thanks @AlexLi0104 .  Looks like travis and appveyor are still unhappy?

AlexLi0104 commented:
@cboettig Indeed. I was thinking maybe we could discuss this a little bit today. I can't seem to figure out what actually went wrong, and every time I think I fixed something more errors appear. My conjecture is that it has something to do with the "overwrite" interaction with user that we just added.

cboettig commented:
yup, we can take a look today.  I have a meeting at 1:45p, do you think you
could come by at 1:20?

Right, it looks to me like you need to be sure to check that the system is
interactive with `interactive()` before asking for input.  The CI systems
are just stuck waiting for input because the function doesn't have the
logic to determine that interactive() is FALSE, and so it should default to
some behavior (e.g. fall back to overwrite=TRUE)

HTH,

Carl
---
Carl Boettiger
http://carlboettiger.info/


On Tue, Mar 19, 2019 at 9:45 AM AlexLi0104 <notifications@github.com> wrote:

> @cboettig <https://github.com/cboettig> Indeed. I was thinking maybe we
> could discuss this a little bit today. I can't seem to figure out what
> actually went wrong, and every time I think I fixed something more errors
> appear. My conjecture is that it has something to do with the "overwrite"
> interaction with user that we just added.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/boettiger-lab/gpdd/pull/23#issuecomment-474463998>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AANletOLniFV-VeTdUfhp5J_nYLjZLg5ks5vYRQygaJpZM4b7TDv>
> .
>

AlexLi0104 commented:
No problem. I’ll be there at 1:20.

Best
Alex

> 在 2019年3月19日，上午10:18，Carl Boettiger <notifications@github.com> 写道：
> 
> yup, we can take a look today. I have a meeting at 1:45p, do you think you
> could come by at 1:20?
> 
> Right, it looks to me like you need to be sure to check that the system is
> interactive with `interactive()` before asking for input. The CI systems
> are just stuck waiting for input because the function doesn't have the
> logic to determine that interactive() is FALSE, and so it should default to
> some behavior (e.g. fall back to overwrite=TRUE)
> 
> HTH,
> 
> Carl
> ---
> Carl Boettiger
> http://carlboettiger.info/
> 
> 
> On Tue, Mar 19, 2019 at 9:45 AM AlexLi0104 <notifications@github.com> wrote:
> 
> > @cboettig <https://github.com/cboettig> Indeed. I was thinking maybe we
> > could discuss this a little bit today. I can't seem to figure out what
> > actually went wrong, and every time I think I fixed something more errors
> > appear. My conjecture is that it has something to do with the "overwrite"
> > interaction with user that we just added.
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/boettiger-lab/gpdd/pull/23#issuecomment-474463998>,
> > or mute the thread
> > <https://github.com/notifications/unsubscribe-auth/AANletOLniFV-VeTdUfhp5J_nYLjZLg5ks5vYRQygaJpZM4b7TDv>
> > .
> >
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

cboettig commented on R/download_gpdd.R:
```diff
@@ -19,17 +20,71 @@ download_gpdd <- function(dataset_name = c("gpdd_data",
     check_name(dataset_name[i])
     if (file.exists(file.path(dir, csv_name))) {
-      message(paste(dataset_name[i], "already exists. Proceeding to the next dataset.\n", sep = ' '))
+      if (overwrite == "Ask") {
```
Try something like: `(overwrite == "Ask" && interactive())`

cboettig commented on R/helper.R:
```diff
@@ -36,3 +36,17 @@ dataset_in_dir <- function(dir = rappdirs::user_data_dir()) {
 }
+
+
+ask_yn <- function(...) {
```
Add some roxygen for this:

```
#' @importFrom crayon green
#' @importFrom utils menu
```

cboettig replied:
You'll also need to run:

```
usethis::use_package("crayon")
usethis::use_package("utils")
```

Or just add crayon and utils to the Imports section of the DESCRIPTION file manually.

cboettig commented:
@AlexLi0104 Looking good.  In the download function, you've repeated the code for overwriting and or not; which you don't want to do.  Note that this also has a big impact on code coverage: https://codecov.io/gh/boettiger-lab/gpdd/src/Alex/R/download_gpdd.R

You want to rework the logic so that you don't have separate paths that do the same thing...

AlexLi0104 commented:
@cboettig I have reworked the logic, so that I don't have repeated chunks of code any more. The code coverage now is about 90%, and those not covered are ones involving interactions. Please feel free to take a look and let me know what else need to be changed. Thanks!

cboettig commented:
@AlexLi0104 Excellent, very nice work.
